test/functional/testSettings.js

```diff
@@ -0,0 +1,590 @@
+    var remote;
+
+    var expectedAlertText = 'Are you sure you want to remove this index pattern?';
+    return {
+      // on setup, we create an settingsPage instance
+      // that we will use for all the tests
+      setup: function () {
+        common = new Common(this.remote);
+        settingsPage = new SettingsPage(this.remote);
+        headerPage = new HeaderPage(this.remote);
+        scenarioManager = new ScenarioManager(url.format(config.elasticsearch));
+        remote = this.remote;
+      },
+
+      beforeEach: function () {
+
+        // start each test with an empty kibana index
+        return scenarioManager.reload('emptyKibana')
+        // and load a minimal set of makelogs data
+        .then(function loadIfEmptyMakelogs() {
+          return scenarioManager.loadIfEmpty('makelogs');
+        })
+        .then(function () {
+          return common.sleep(2500);
+        })
+        .then(function () {
+          return common.tryForTime(25000, function () {
+            return remote.get(url.format(_.assign(config.kibana, {
+              pathname: ''
+            })))
+            .then(function () {
+              return common.sleep(2000);
+            })
+            .then(function () {
+              return remote.getCurrentUrl()
+              .then(function (currentUrl) {
+                expect(currentUrl).to.contain('settings');
+              });
+            });
+          });
+        });
+      },
+
+      teardown: function unloadMakelogs() {
+        return scenarioManager.unload('makelogs');
+      },
+
+      /*
+       ** Test the default state of checboxes and the 2 text input fields
+       */
+      testSettingsInitialState: function () {
+        var testSubName = 'testSettingsInitialState';
+
+        return settingsPage.getTimeBasedEventsCheckbox().isSelected()
+        .then(function (selected) {
+          expect(selected).to.be.ok();
+        })
+        .then(function () {
+          return settingsPage.getNameIsPatternCheckbox().isSelected()
+          .then(function (nameIsPatternSelected) {
+            expect(nameIsPatternSelected).to.not.be.ok();
+          });
+        })
+        .then(function () {
+          return settingsPage.getIndexPatternField()
+          .then(function (patternField) {
+            return patternField.getProperty('value')
+            .then(function (pattern) {
+              common.log('patternField value = ' + pattern);
+              expect(pattern).to.be('logstash-*');
+            });
+          });
+        })
+        .then(function () {
+          return settingsPage.getTimeFieldNameField().isSelected()
+          .then(function (timeFieldIsSelected) {
+            common.log('timeField isSelected = ' + timeFieldIsSelected);
+            expect(timeFieldIsSelected).to.not.be.ok();
+          });
+        })
+        .catch(function screenshotError(reason) {
+          return common.screenshotError(testSubName, reason);
+        });
```

Comment:
There are many nested `then`s here. Is it possible to flatten the hierarchy? 

```javascript
return settingsPage.getTimeBasedEventsCheckbox().isSelected()
.then(function (selected) {
  expect(selected).to.be.ok();
})
.then(function () {
  return settingsPage.getNameIsPatternCheckbox().isSelected()
})
.then(function (nameIsPatternSelected) {
  expect(nameIsPatternSelected).to.not.be.ok();
});
.then(function () {
  return settingsPage.getIndexPatternField()
})
.then(function (patternField) {
  return patternField.getProperty('value')
})
.then(function (pattern) {
  common.log('patternField value = ' + pattern);
  expect(pattern).to.be('logstash-*');
})
.then(function () {
  return settingsPage.getTimeFieldNameField().isSelected()
})
.then(function (timeFieldIsSelected) {
    common.log('timeField isSelected = ' + timeFieldIsSelected);
    expect(timeFieldIsSelected).to.not.be.ok();
  });
})
.catch(function screenshotError(reason) {
  return common.screenshotError(testSubName, reason);
});
```

I prefer only nesting `then